include/net/ieee802154_radio.h

```diff
@@ -178,6 +185,18 @@ struct ieee802154_config {
 
 		/** ``IEEE802154_CONFIG_EVENT_HANDLER`` */
 		ieee802154_event_cb_t event_handler;
+
+		/** ``IEEE802154_CONFIG_MAC_KEYS`` */
+		struct {
+			uint8_t key_id_mode;
+			uint8_t key_id;
+			uint8_t *prev_key;
+			uint8_t *curr_key;
+			uint8_t *next_key;
```

Comment:
Are the driver changes that use this data coming later?
I was also wondering about these pointer values, these probably only work properly if the driver is copying the data to some safe place.

Reply:
Yes, you are right. This PR implements only the API, the driver part should come later on. The idea is that the driver copies the keys for its internal usage.